mtags/src/main/scala-3/scala/meta/internal/pc/completions/OverrideCompletions.scala

```diff
@@ -172,4 +324,50 @@ object OverrideCompletions:
 
   private val interestingFlags = Flags.Method | Flags.Mutable
 
+  /**
+   * Infer the editPosition for "implement all" code action for the given TypeDef.
+   *
+   * If there're braces like `class FooImpl extends Foo {}`,
+   * editPosition will be inside the braces.
+   * Otherwise, e.g. `class FooImpl extends Foo`, editPosition will be
+   * after the `extends Foo`.
+   *
+   * @param text the whole text of the source file
+   * @param td the class/object to impement all
+   */
+  private def inferEditPosition(text: String, td: TypeDef)(using
+      Context
+  ): SourcePosition =
+    val span = hasBraces(text, td)
+      .map { offset =>
+        td.sourcePos.span.withStart(offset + 1).withEnd(offset + 1)
+      }
+      .getOrElse(
+        td.sourcePos.span.withStart(td.span.end)
+      )
+    td.sourcePos.withSpan(span)
+  end inferEditPosition
+
+  private def hasBraces(text: String, td: TypeDef): Option[Int] =
+    def hasSelfTypeAnnot = td.rhs match
+      case t: Template =>
+        t.self.span.isSourceDerived
+      case _ => false
+    val start = td.span.start
+    val marker = if hasSelfTypeAnnot then '>' else '{'
+    val offset = text.indexOf(marker, start)
```

Comment:
What if there is a comment with `\\ >` ? Are we sure that `=>` will be the first thing to find?

Reply:
fixed :+1: https://github.com/scalameta/metals/pull/3960/commits/ea6a10b7b4aca3b11ca1eee234f897cda7f63bd8